ambry-replication/src/main/java/com/github/ambry/replication/continuous/DataNodeTracker.java

```diff
@@ -77,11 +83,68 @@ public DataNodeId getDataNodeId() {
     return dataNodeId;
   }
 
+  public List<GroupTracker> getGroupTrackers() {
+    List<GroupTracker> groupTrackers = new ArrayList<>(activeGroupTrackers);
+    groupTrackers.add(standByGroupTracker);
+    return groupTrackers;
+  }
+
   public List<ActiveGroupTracker> getActiveGroupTrackers() {
     return activeGroupTrackers;
   }
 
   public StandByGroupTracker getStandByGroupTracker() {
     return standByGroupTracker;
   }
+
+  /**
+   * @return returns remote replica groups that are getting tracked in group trackers of
+   * this data node tracker
+   */
+  public List<ReplicaThread.RemoteReplicaGroup> getInflightRemoteReplicaGroups() {
+    return getGroupTrackers().stream()
+        .filter(GroupTracker::isInFlight)
+        .map(GroupTracker::getRemoteReplicaGroup)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Iterates over all group trackers and if remote replica group is getting tracked by group tracker is done,
+   * removes it from group tracker
+   * @return returns list of remote replica groups removed from tracking
+   */
+  public List<ReplicaThread.RemoteReplicaGroup> processFinishedGroups() {
+    List<ReplicaThread.RemoteReplicaGroup> finishedGroups = new ArrayList<>();
+    getGroupTrackers().forEach(groupTracker -> {
+      if (!groupTracker.isGroupDone()) {
+        return;
+      }
+      finishedGroups.add(groupTracker.getRemoteReplicaGroup());
+      groupTracker.finishIteration();
```

Comment:
When group is already done then why do we need to call `finishIteration()`

Reply:
finishIteration removed RemoteReplicaGroup from group tracker and clears replica trackers List.

In a group tracker we are making remotereplicagroup as null too also we are updating replica tracker for throtlling and status.